src/com/xilinx/rapidwright/router/VersalClockRouting.java

```diff
@@ -102,65 +127,86 @@ public static Node routeBUFGToNearestRoutingTrack(Net clk) {
         return null;
     }
 
+    private static final String VROUTE = "VROUTE";
+    
+    public static Integer getRoutingTrackIndex(Node node) {
+        if (node.getIntentCode() == IntentCode.NODE_GLOBAL_VROUTE) {
+            String wireName = node.getWireName();
+            return Integer.parseInt(wireName.substring(wireName.indexOf(VROUTE) + VROUTE.length()));
+        }
+        return null;
+    }
+    
+    private static boolean isCentroidCandidateAvailable(Node node, Set<Integer> unavailableTracks) {
```

Comment:
Is it insufficient to use `getNodeStatus()` to identify unavailable tracks? As I understand, after routing one clock, all its nodes gets preserved and thus blocked for use by anyone else.